### PR TITLE
Bump rust-toolchain to 1.77.2

### PR DESCRIPTION
**Pull request by mpapierski**
Update rust-toolchain to 1.77.2. In an attempt to shrink down #4806 which requires 1.74.0 stable, I tried to upgrade as high as I could and we're just two versions behind (1.79 is currently stable). I tried to go to the current stable, but the amount of extra warnings (tons of `field x is never read` etc.) made me uncomfortable pursuing this further.

Once merged it should slightly decrease the size of #4806 PR.

### CODE REVIEW

File: node/src/components.rs [hunk 1]
```diff
@@ -155,6 +155,7 @@ pub(crate) trait InitializedComponent<REv>: Component<REv> {
         self.state() == &ComponentState::Uninitialized
     }
 
+    #[allow(dead_code)]
```
Comment (rafal-ch) [inline node/src/components.rs]: Shall we just get rid of the function?

File: node/src/types/exit_code.rs [hunk 2]
```diff
@@ -11,18 +11,14 @@ const SIGNAL_OFFSET: u8 = 128;
...
 pub enum ExitCode {
     /// The process should exit with success.  The launcher should proceed to run the next
     /// installed version of `casper-node`.
     Success = 0,
     /// The process should exit with `101`, equivalent to panicking.  The launcher should not
     /// restart the node.
     Abort = 101,
-    /// The process should exit with `102`.  It used to be an indication to the launcher
-    /// that it should proceed to run the previous installed version of `casper-node`.
-    /// It is no longer used, but we keep it here to avoid it being reassigned to other features.
```
Comment (rafal-ch) [inline node/src/types/exit_code.rs]: I would advise on keeping the variant for the reason mentioned in the comment.

**Review by rafal-ch — APPROVED** [review]
LGTM :+1:

File: execution_engine/src/runtime_context/mod.rs [hunk 3]
```diff
@@ -1274,7 +1274,7 @@ where
                     "Contract".to_string(),
                     other.type_name(),
                 ))),
-                None => Err(TrackingCopyError::KeyNotFound(key)).map_err(Into::into),
+                None => Err(Into::into(TrackingCopyError::KeyNotFound(key))),
```
Comment (jacek-casper) [inline execution_engine/src/runtime_context/mod.rs]:
```suggestion
                None => Err(TrackingCopyError::KeyNotFound(key).into()),
```

**Review by jacek-casper — APPROVED** [review]
left 1 small comment but lgtm other than that

File: node/src/components.rs (hunk 1 above)
Comment (mpapierski) [reply node/src/components.rs]: Removed

File: node/src/types/exit_code.rs (hunk 2 above)
Comment (mpapierski) [reply node/src/types/exit_code.rs]: Done

File: execution_engine/src/runtime_context/mod.rs (hunk 3 above)
Comment (mpapierski) [reply execution_engine/src/runtime_context/mod.rs]: Done

**Pull request merged**